libs/vault/src/cipher-view/cipher-view.component.ts

```diff
@@ -152,7 +154,12 @@ export class CipherViewComponent implements OnChanges, OnDestroy {
 
     const userId = await firstValueFrom(this.activeUserId$);
 
-    if (this.cipher.edit && this.cipher.viewPassword) {
+    // Show Tasks for Manage and Edit permissions
+    // Using cipherService to see if user has access to cipher in a non-AC context to address with Edit Except Password permissions
+    const allCiphers = await firstValueFrom(this.cipherService.ciphers$(userId));
+    const cipherServiceCipher = allCiphers[this.cipher?.id as CipherId];
+
+    if (cipherServiceCipher.edit && cipherServiceCipher.viewPassword) {
```

Comment:
When `cipher?.id` evaluates to `undefined` from the optional chaining `cipherServiceCipher` can become undefined, which would error out on line `162`.

All that to say, I think you should add a check for it just to be safe:

```suggestion
    const cipherServiceCipher = allCiphers[this.cipher?.id as CipherId];

    if (cipherServiceCipher && cipherServiceCipher.edit && cipherServiceCipher.viewPassword) {
```

Reply:
Agreed! Also, this will absolutely happen when an admin is viewing a cipher they do not normally have access to within the AC (the `cipherService` will never have such a cipher).